### PR TITLE
Added possibility to specify Volatility profile per machine

### PR DESCRIPTION
**Pull request by SteveClement**
Added a profile option in the config file (virtualbox.conf) to specify a volatility profile, bypassing auto-detection and speeding up the analysis of the submissions.

### CODE REVIEW

**Comment by KillerInstinct** [comment]
I took a stab at this a while back also, feel free to use/refence:
https://github.com/KillerInstinct/cuckoo-modified/commit/0b925f567e05535f6ce9c5ecc1d1e211f9c8d39d

**Comment by botherder** [comment]
Your implementation looks good. I also don't see the need to store the information in the DB.

**Comment by jbremer** [comment]
I'd be in favor of propagating this information through the `options` field. This should allow to replace the somewhat ugly `profile = Config(manager).get(vm).get("mem_profile")` by something such as `self.task.options["volprofile"]`. Thoughts?

**Comment by jbremer** [comment]
Given the *size* of this feature - considering volatility is a pretty addition to Cuckoo, I'd propose to add a column to the `machines` table to reflect the volatility profile. That'd make @KillerInstinct's patch a bit more properly developed.

File: modules/processing/memory.py [hunk 1]
```diff
@@ -929,9 +932,25 @@ def __init__(self, memfile, osprofile=None):
             if pid:
                 self.mask_pid.append(int(pid))
 
+        response = {}
+
+        task = db.view_task(db.list_tasks(limit=1)[0].id, details=True)
```
Comment (jbremer) [inline modules/processing/memory.py]: What is happening here; getting an instance of the current task? If so we can just replace this with `self.task`.

**Comment by jbremer** [comment]
I'll put this on my personal todo list for the upcoming months (as I'll also be using this functionality), but for reasons stated above I won't be merging this PR. So will close it until I find the time to properly implement it, perhaps also based on @KillerInstinct's commit(s). Even though it's been almost 2 years now, thanks for your contributions ;-)

**Comment by SparkyNZL** [comment]
that would be awesome, I have mine default to Win7 but its not that good
for WinXP or Win10 :(

**Comment by jbremer** [comment]
For what its worth, support for per-VM `osprofile`'s has been released as part of version `2.0.2`.

**Pull request closed without merging**